franklin/github/views.py

```diff
@@ -55,6 +63,152 @@ def callback(request):
 
     return HttpResponse(status=200)
 
+def make_rest_get_call(url, headers):
+    response = None
+    try:
+        response = requests.get(url, headers=headers)
+    except (ConnectionError, HTTPError, Timeout) as e:
+        logger.error('REST GET Connection exception : %s', e)
+    except:
+        logger.error('Unexpected REST GET error: %s', sys.exc_info()[0])
+
+    if response is not None:
+        if status.is_success(response.status_code):
+            return response
+        else:
+            logger.warn('Bad GET response code of', response.status_code)
+    return None
+
+def make_rest_post_call(url, headers, body):
+    response = None
+    try:
+        response = requests.post(url, data=json.dumps(body), headers=headers)
+    except (ConnectionError, HTTPError, Timeout) as e:
+        logger.error('REST POST Connection exception : %s', e)
+    except:
+        logger.error('Unexpected REST POST error: %s', sys.exc_info()[0])
+
+    if response is not None:
+        if status.is_success(response.status_code):
+            return response
+        else:
+            logger.warn('Bad POST response code of', response.status_code)
+    return None
+
+# Not currently used. but it works.
+#def get_owners_repos(owner_name):
+#    have_next_page = True
+#    url = github_base + 'orgs/' + owner_name + '/repos?per_page=100'
+#    # TODO - Confirm that a header token is the best/most secure way to go
+#    headers = {
+#                'content-type': 'application/json',
+#                'Authorization': 'token ' + oauth
+#              }
+#    repos = []
+#
+#    while have_next_page:
+#        response = None
+#        have_next_page = False # when in doubt, we'll leave the loop after 1
+#        response = make_rest_get_call(url, headers)
+#
+#        if response is not None:
+#            # Add all of the repos to our list
+#            for repo in response.json():
+#                repo_data = {}
+#                repo_data['id'] = repo['id']
+#                repo_data['name'] = repo['name']
+#                repos.append(repo_data)
+#
+#            # If the header has a paging link called 'next', update our url
+#            # and continue with the while loop
+#            if response.links and response.links.get('next', None):
+#                url = response.links['next']['url']
+#                have_next_page = True
+#
+#    if not repos:
+#        logger.error('Failed to find repos for owner', owner_name)
+#    return repos
+
+def get_site_from_request(request):
+    owner_name = request.data.get('owner', None)
+    owner_id = request.data.get('owner_id', None)
+    repo_name = request.data.get('repository', None)
+    repo_id = request.data.get('repository_id', None)
```

Comment:
Agreed. The other serializer that you wrote was awesome, would love to see more of that